### DISCUSSION TITLE
Help for a beginner, related to CFG Unlock

### DISCUSSION THREAD

**Discussion by stoivo** [hackintosh-club/intel-nuc10#70] [Q&A]
Hi, I am new to Hackintosh so please beer with me.

Related to CFG Unlock. Should I follow the standard guide at https://dortania.github.io/OpenCore-Post-Install/misc/msr-lock.html? or should I do the thing in the readme, if so where do I getn `CFGLock Shell.efi`?

**Comment by ghost** [comment]
Boot the USB drive with the opencore EFI folder copied into the EFI partition and press the space bar when you see the opencore bootloader. You'll see the CFGLock Shell option.

If you need to know how to create the USB drive. Disk Utility will automatically create the required EFI partition.
https://onmac.net/how-to-make-a-macos-bootable-usb/

You may need to revert pack to BIOS 0055 for CFGLock to work, then install MacOS and lastly update to the latest bios.
https://github.com/mackonsti/nuc10i7fnh/tree/master/BIOS

**Comment by stoivo** [comment]
I just want to say it write ahead. Thank you, really appreciate the help. 

I installed BIOS 0055. Reformated the USB drive and, installed it with a Monterey installer. Downloaded this project's latest release and copied it to the EFI partition. Started the machine from the USB and found the `CFGLock Shell.efi` I managed to unlock the CFG. I added `ControlMsrE2` to verify that I had disabled it, found it here https://dortania.github.io/OpenCore-Post-Install/misc/msr-lock.html#checking-if-your-firmware-supports-cfg-lock-unlocking.

Right now the installer is running.

I was confused by the instruction from the readme. 

---

### CFG Unlock

If it's the first time your nuc10 install with macOS, you must unlock CFG before install macOS.


1. Find a USB flash drive, formatted as `FAT32` partition.
2. Copy `EFI` files to the root of your USB drive, path will be `/EFI`.
3. Reboot your nuc10, press `F10`, boot with the drive above. 
4. Choose `CFGLock Shell.efi` unlock CFG. 

**When in the OC's booting interface, press the `space bar` to show more options**

---

I was asking myself should I follow this other guide on creating the installer or should I format it FAT32 as it said here? Maybe we can update the readme so it's easier to understand?


What about?
---

### CFG Unlock

If it's the first time you install macOS on your nuc10, you must unlock CFG before installing macOS.

1. Create the installer USB, follow on of these guides
  opencore Guide: https://dortania.github.io/OpenCore-Install-Guide/installer-guide/
  onmoc.net https://onmac.net/how-to-make-a-macos-bootable-usb/
2. Copy `EFI` files from this repository into UFI partition on your USB drive.
  Use EFI Agent or MountEFI to mount the EFI partition.
3. Reboot your nuc10, press `F10`, boot with the drive above. 
4. Choose `CFGLock Shell.efi` unlock CFG. 

**When in the OC's booting interface, press the `space bar` to show more options**

**Comment by stoivo** [comment]
Right now I am having a hard time detecting my NVME harddrive. I get to the Disk Util but my nvme is not showing up. I hadden to have a `Samsung 970 EVO Plus NVMe` which in know to have some issues https://dortania.github.io/OpenCore-Install-Guide/macos-limits.html#storage-support

Any tips?

**Comment by ghost** [comment]
Did you update the SSD firmware?
https://semiconductor.samsung.com/resources/user-manual/Firmware_Update_Utility_UserManual.pdf
https://semiconductor.samsung.com/resources/software-resources/Samsung_SSD_970_EVO_Plus_4B2QEXM7.iso

Did you update the BIOS to 0057 after unlocking the CFG lock? I would also try this after the SSD firmware update. I'm running 0057 without issues.

Also make sure that the SSD is detected in the BIOS.

The guide on the front page is only for unlocking the CFG lock. It is better to just create the MacOS install bootable USB.

The best way of creating a MacOS install USB drive with the required EFI partition is per the guide I listed. Then all we need to do is mount the EFI partition with the [EFI Agent](https://github.com/headkaze/EFI-Agent) app and lastly copy the EFI folder to the EFI partition. Note: The EFI folder also needs to be copied to the EFI partition of the main SSD drive right after the first MacOS boot. It is important to copy the EFI zip file and EFI Agent app to the USB drive with the MacOS install app file.

Choosing the Mac OS Extended (Journaled) format and the `GUID Partition Map scheme` in Disk Utility will lead to Disk Utility creating a 200MB FAT32 partition for EFI folder at the beginning of the USB drive and the rest of the drive will be Mac OS Extended (Journaled) for the Install Monterey app file. Same thing occurs when we select APFS with `GUID Partition Map scheme` during MacOS installation.

Let me know if you're still having issues.

**Reply by stoivo** [reply]
Updated bios to 0057, no change. NVME is detected in BIOS

My firmmware nvme : 2B2QEXM7 

It is the latest version I can install. 3B2QEXM7 and 4B2QEXM7 is also released for 970 EVO Pluss but Samsung made a revision and changed a controller so I can't install these updated on my NVMe because it has the old controller.

https://www.reddit.com/r/pcmasterrace/comments/pmvc9k/samsung_970_evo_plus_ssd_hardware_revisions_does/
https://www.reddit.com/r/pcmasterrace/comments/pmvc9k/comment/hocexic/?utm_source=share&utm_medium=web2x&context=3

Any suggestions?

I might just install a SSH too and use it as boot device and mount a partition from my nvme as /Users

**Reply by ghost** [reply]
The following link surprisingly works. I changed the 4 to a 2. See if this update fixes the issues. Firmware date is 2019-04-24.
https://semiconductor.samsung.com/resources/software-resources/Samsung_SSD_970_EVO_Plus_2B2QEXM7.iso

Article from May 26, 2019 and therefore the firmware above should work.
https://bartechtv.com/samsung-releases-new-firmware-for-970-evo-plus-that-resolves-macos-compatibility-issues/

Additional ISO link from article above.
https://s3.ap-northeast-2.amazonaws.com/global.semi.static/SAMSUNG_SSD_970EVO_Plus_ISO_190513/7ES357322A6707A720E1A71EF11A3BE1EED819E011D317626415F0281A78151C/Samsung_SSD_970_EVO_Plus_2B2QEXM7.iso

If the update above doesn't work. I'd suggest installing Windows 11 with the 30 day trial and then Samsung Magician to see if there's an update via Samsung Magician.
https://www.google.com/search?q=download+windows+11
https://semiconductor.samsung.com/consumer-storage/support/tools/

Please make sure to install an NVME SSD thermal pad if you haven't. 20x67x1.0mm if all the SSD chips are even.

I'm running the 970 Pro and I'm glad I spent the extra money because it is well worth it. 980 Pro is available now.

Did you try with and without the NVMeFix.kext? I don't believe it's required if the firmware is up to date.

**Reply by stoivo** [reply]
Thanks for all the research.

My nvme drive was already at 2B2QEXM7, so no luck there. I had also tried installing windows 11 and use Samsung Magician, it said I was on the latest release.

I don't know what happen, but my nothing happend when I click the power button. I talked with my supplier and they want it it return and can send me a new one. I considering changing the NVME drive too. I am thinking of Samsung 970 Pro or Samsung 980 Pro, thoughts?

Termal pad. I think my came with it on. This blue pad
![20220719_235708](https://user-images.githubusercontent.com/3492040/180382575-93ccde19-0198-45db-a009-bacfe8d645ea.jpg)

**Reply by ghost** [reply]
That's good news.

Chips on Samsung SSDs have different heights and require different height thermal pads. I installed 0.5mm pad on the 3rd chip and 1mm pad atop the whole Samsung 970 Pro SSD. Similar to this video https://www.youtube.com/watch?v=I8Z09nU554Q

I purchased a similar multi height pack in grey. https://www.amazon.ca/dp/B09C5PDZ1Q

https://www.reddit.com/r/hardware/comments/jebhd1/970_pro_vs_980_pro_discussion/
https://premiumbuilds.com/comparisons/samsung-980-pro-vs-970-pro/
https://ssd.userbenchmark.com/Compare/Samsung-980-Pro-NVMe-PCIe-M2-500GB-vs-Samsung-970-Pro-NVMe-PCIe-M2-512GB/m1307906vsm498971

**Reply by stoivo** [reply]
I sounds like I should go for the 1 TB samsung 970. I will check what other drives area availiable too.

**Reply by stoivo** [reply]
sorry, I miss read. I don't see any reason why not to pick a 980 If am changing.

**Reply by stoivo** [reply]
I am also considering using a Kingston FURY Renegade 2TB SSD, https://www.multicom.no/business/kingston-fury-renegade-2tb-ssd/cat-p/c/p1002360550.

I know it is PCIe 4.0 but the nuc is PCIe 3.0 but it should work. What do you think?

**Reply by stoivo** [reply]
I give up trying to setup Hackintosh on this NUC with this disk. Thanks for all the help @essereio

**Reply by ghost** [reply]
Which disk did you ended up buying?

**Reply by stoivo** [reply]
I didn't buy a new one. I was't able to return it and then my NUC die so I had to return the NUC and 2 months later I got it back, but in those two months I got a new work laptop M1... And I did't want to speend more time trying to setup Hackintosh. I father use it for daily driving and I use it for fileserver
